app/query_objects/base_filtered_query.rb

```diff
@@ -0,0 +1,16 @@
+class BaseFilteredQuery
+  def initialize(relation, filter_params = {})
+    @relation = relation
+    @filter_params = filter_params
+  end
+
+  def all
+    filter_params.reduce(relation) do |relation, (key, value)|
+      send("by_#{key}", relation, value)
```

Comment:
I have made a typo. Should we use `public_send` here? I do not think that all methods should be private in query objects.
```suggestion
      public_send("by_#{key}", relation, value)
```